app/components/post_textbox/post_textbox.js

```diff
@@ -451,6 +458,35 @@ export default class PostTextbox extends PureComponent {
         });
     };
 
+    onCloseChannelPress = () => {
+        const {defaultChannel, channelId} = this.props;
+        const {setChannelDisplayName, setChannelLoading} = this.props.actions;
+        setChannelLoading(true);
+        setChannelDisplayName(defaultChannel.display_name);
+        EventEmitter.emit('switch_channel', defaultChannel, channelId);
+    };
+
+    archivedView = (theme, style) => {
+        return (<View style={style.archivedWrapper}>
```

Comment:
I thought we talked about having a component for this, then the `onCloseChannelPress` and the `defaultChannel` can be part of that component props? as well as the actions ;)

Reply:
We did but I'm not clear on why you would prefer that; what was your reasoning?

Reply:
simplicity and maintainability?

Reply:
Please elaborate specifically.

Reply:
ok so the postTextBox component is already huge and with this addition is even longer and now it even receives more props and listens to more redux changes, I do believe that splitting the component into smaller components makes it easier to maintain and makes the code more readable, I do get that this works just fine, but yet again making the app as a single component will do just fine as well.

The archived view should be very rarely seen so I think it should be in its own component if possible.

Reply:
The archived message must replace *all* instances of the post input when the channel is archived. We will never show just use the `PostTextbox` without the possibility of it being replaced by the archived message, nor would we show both of them together. That's why I put them together.

Reply:
Hmmm, for example to me having the post textbox render but in readonly mode makes sense to have it on the same component as it renders the post texbox but having it replaced with another thing it doesn’t make sense.

It should render either componentA or componentB.

Something like a `ChannelFooter` component that decides if you render the texbox or the archived component.

Makes sense?

Reply:
I prefer it as part of the component so since it's really just a matter of opinion I'm going to invoke the "opinion goes to the developer" and leave it as is. Feel free to refactor it if you'd like afterwards if you still feel strongly about it.

Reply:
lol ok